src/linalg_traits.rs

```diff
@@ -65,5 +68,36 @@ pub trait NdFloat:
 {
 }
 
+#[cfg(feature = "std")]
 impl NdFloat for f32 {}
+#[cfg(feature = "std")]
 impl NdFloat for f64 {}
+
+/// A no_std floating-point element type represented by `f32` and `f64`.
+///
+/// Trait `NdFloatCore` is only implemented for `f32` and `f64` and encompasses
+/// much float-relevant ndarray functionality as possible, while working for no_std
+/// code, including the traits needed for linear algebra and for *right hand side* scalar
+/// operations (`ScalarOperand`).
+///
+/// This trait can only be implemented by `f32` and `f64`.
+pub trait NdFloatCore:
```

Comment:
I'd propose that we don't add a new NdFloatCore trait (The story behind NdFloat itself is fuzzy as it is) - we don't need it.

Reply:
So just remove NdFloat then?

Reply:
ndarray-linalg relies on NdFloat and maybe some other ndarray using crates.. I'm not sure about just removing it. But I'll remove `NdFloatCore`